nodejs/week2/app.js

```diff
@@ -0,0 +1,88 @@
+import express, { query } from "express";
+import fs from "fs";
+import path from "path";
+
+const app = express();
+const port = process.env.PORT || 3000;
+
+// Support parsing JSON requests
+app.use(express.json());
+
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+
+app.listen(port, () => {
+  console.log(`Listening on port ${port}`);
+});
+
+app.get("/search", (req, res) => {
+  const query = req.query.q; // Convert query to lowercase
+  const jsonData = fs.readFileSync("./document.json");
+  const data = JSON.parse(jsonData);
+
+  if (!query) {
+    return res.send(data);
+  }
+
+  let found = false;
+
+  const matchingData = data.filter(
+    (element) =>
+      String(element.value).toLowerCase().includes(query.toLowerCase()) ||
+      String(element.description).toLowerCase().includes(query.toLowerCase())
+  );
+
+  if (matchingData.length > 0) {
+    res.send(matchingData);
+    found = true;
+  }
+  if (!found) {
+    res.status(404).send("There is an error in fetching data");
+  }
+});
+
+app.get("/document/:id", (req, res) => {
+  const param = req.params.id;
+  try {
```

Comment:
I can see that this function has error handling with try-catch, which is good, but some other functions don't have it. It's better to be consistent and implement it for all of them 😉